plugins/inputs/win_wmi/README.md

```diff
@@ -0,0 +1,280 @@
+# Windows Management Instrumentation Input Plugin
+
+This document presents the input plugin to read WMI classes on Windows
+operating systems. With the win_wmi plugin, an administrator is enabled to
+capture and filter virtually any configuration or metric value exposed through
+the Windows Management Instrumentation
+([WMI](https://learn.microsoft.com/en-us/windows/win32/wmisdk/wmi-start-page))
+service.
+
+If telegraf is configured with a logfile and the plugin's configuration
+contains an invalid namespace, class, or property, an error is logged.
+
+## Global configuration options <!-- @/docs/includes/plugin_config.md -->
+
+In addition to the plugin-specific configuration settings, plugins support
+additional global and plugin configuration settings. These settings are used to
+modify metrics, tags, and field or create aliases and configure ordering, etc.
+See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
+
+[CONFIGURATION.md]: ../../../docs/CONFIGURATION.md
+
+## Configuration
+
+```toml @sample.conf
+# Input plugin to query Windows Management Instrumentation
+[[inputs.win_wmi]]
+  # specifies a prefix to attach to the measurement name
+  name_prefix = "win_wmi_"
+  [[inputs.win_wmi.query]]
+    # a string representing the WMI namespace to be queried
+    namespace = "root\\cimv2"
+    # a string representing the WMI class to be queried
+    class_name = "Win32_Volume"
+    # an array of strings representing the properties of the WMI class to be queried
+    properties = ["Name", "Capacity", "FreeSpace"]
+    # a string specifying a WHERE clause to use as a filter for the WQL
+    filter = 'NOT Name LIKE "\\\\?\\%"'
+    # WMI class properties which should be considered tags instead of fields
+    tag_properties = ["Name"]
+```
+
+### namespace
+
+A string representing the WMI namespace to be queried. For example,
+`root\\cimv2`.
+
+### class_name
+
+A string representing the WMI class to be queried. For example,
+`Win32_Processor`.
+
+### properties
+
+An array of strings representing the properties of the WMI class to be queried.
+
+### filter
+
+A string specifying a WHERE clause to use as a filter for the WMI Query
+Language (WQL).
+
+### tag_properties
+
+Properties which should be considered tags instead of fields.
+
+## Metrics
+
+By default, a WMI class property's value is used as a metric field. If a class
+property's value is specified in `tag_properties`, then the value is
+instead included with the metric as a tag.

```

Comment:
What happens if a user specifies all the returned values as tags? 

Better yet, why not make values that are strings tags by default?

Reply:
If every property is specified as a tag, then there's no metric to return. A tags map and an empty fields map are passed to the accumulator's AddFields() method. The accumulator appears to silently continue without error. I've only ever used the prometheus output plugin, so this is a non-issue for me since prometheus only accepts packages with valid metrics.

The initial draft of this plugin made all strings tags, but the feedback was to allow the user to specify what is considered a tag. There's also the off chance a metric is presented as a string, so auto-detection could be unreliable.

Reply:
> then there's no metric to return

Right this is what I expect, but I want to be certain users know this. Can you please add a note about this situation.

Reply:
> The initial draft of this plugin made all strings tags, but the feedback was to allow the user to specify what is considered a tag. 

After sleeping on this, I think this is fine as well.

Reply:
This isn't any different to like the `csv_tag_columns` parameter which also doesn't have this note..

Reply:
yep and it should